src/service/repo/anonymousMessageRepo.ts

```diff
@@ -79,9 +79,13 @@ async function sendAnonymousMessage(
   }
 }
 
-async function getAllAnonymousChannels() {
+async function getAllAnonymousChannels(timeStamp: string | undefined) {
+  // get time stamp
+  const url = timeStamp
+    ? `${baseUrl.getAllAnonymousChannels}?last_fetch_date=${timeStamp}`
+    : baseUrl.getAllAnonymousChannels;
   try {
-    const response = await anonymousApi.get(baseUrl.getAllAnonymousChannels);
+    const response = await anonymousApi.get(url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change introduces a conditional URL construction based on whether a `timeStamp` is provided or not. This is a good practice as it allows for more flexible API calls. However, please ensure that the server-side API supports this query parameter and handles it correctly.



<!-- This is an auto-generated comment by OSS CodeRabbit -->